src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/PropertyBasedEmbeddingServiceConfigStore.java

```diff
@@ -49,6 +49,14 @@ public EmbeddingServiceConfigStore.ServiceConfig getConfiguration(
         throw new JsonApiException(
             ErrorCode.VECTORIZE_SERVICE_TYPE_NOT_ENABLED,
             ErrorCode.VECTORIZE_SERVICE_TYPE_NOT_ENABLED.getMessage() + serviceName);
+      case "cohere":
+        if (config.cohere().enabled()) {
+          return ServiceConfig.provider(
+              serviceName, serviceName, config.cohere().apiKey(), config.cohere().url().toString());
+        }
+        throw new JsonApiException(
```

Comment:
There's new method in `ErrorCode` to simplify throwing:

    throw ErrorCode.VECTORIZE_SERVICE_TYPE_NOT_ENABLED.toApiException(serviceName);

it will call `String.format()`on arguments, concatenate that message to base message defined for enum. Slightly less code to write.

Reply:
Good change will fix it